### PR TITLE
feat: avalanche assets

### LINKED ISSUE #807: Avalanche asset support
### Overview

We need the asset service to support assets on Avalanche C Chain

### References and additional details

* `packages/asset-service`

### Acceptance Criteria

* We have erc20s on Avalanche in the asset service

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

4 hours

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
this adds a subset of supported assets on avalanche

this only includes the ".e" assets, which are erc20s that have been bridge from ethereum to avalance via the latest avalanche-bridge, explained here https://docs.penguinfinance.io/getting-started/new-.e-tokens-explainer

note - this is *not* exhaustive, and can easily be updated at a later date to include more tokens that were previously bridge to avalanche via the legacy bridge, as documented here https://github.com/ava-labs/bridge-tokens

closes #807

### CODE REVIEW

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:40]:
```suggestion
```

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:41]: `toChainId()`

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:50]: comment over todo if this is happening during gen

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:71]: Missed creating the `ASSET_REFERENCE` for avalanche

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:72]: Worth just using the `avalancheAssetId` constant?

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:13]: opinion: I like `Record<>` over the old style syntax

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:24]: moar record! haha

File: packages/asset-service/src/generateAssetData/avalanche/avalanche.ts
Comment (kaladinlight) [inline packages/asset-service/src/generateAssetData/avalanche/avalanche.ts:57]: i find it silly to add the extra vars just for the js key=value syntax when not used elsewhere. more readable imo to have the value being defined in the object so you don't have to look for where it was defined somewhere else

**Review by kaladinlight — APPROVED** [review]
<3

**Review by gomesalexandre — APPROVED** [review]
LGTM, added a non-blocking comment about commenting what's the key of `AvalancheTokenList` type

**Comment by 0xdef1cafe** [comment]
waiting to be able to link to web before i merge this; i'm not sure what will happen on the asset pages w.r.t send/receive.

**Comment by 0xdef1cafe** [comment]
do not merge until we change this to point at coingecko - they have a full list.

File: packages/asset-service/src/generateAssetData/baseAssets.ts
Comment (0xdef1cafe) [inline packages/asset-service/src/generateAssetData/baseAssets.ts:5]: stylistically prefer the plain import but obviously fine

File: packages/asset-service/src/generateAssetData/coingecko.ts
Comment (0xdef1cafe) [inline packages/asset-service/src/generateAssetData/coingecko.ts:61]: we probably want to do a `toAssetId` with the `token.address` before assuming it's the correct override asset - there may be cases where the contract address is the same across layers?

File: packages/caip/src/assetId/assetId.ts
Comment (0xdef1cafe) [inline packages/caip/src/assetId/assetId.ts:26]: @0xApotheosis thoughts on this change?

File: packages/asset-service/src/generateAssetData/baseAssets.ts
Comment (kaladinlight) [reply packages/asset-service/src/generateAssetData/baseAssets.ts:5]: I was just saving some lines so it didn't split import across lines, could update, not a big deal for me either way

File: packages/asset-service/src/generateAssetData/coingecko.ts
Comment (kaladinlight) [reply packages/asset-service/src/generateAssetData/coingecko.ts:61]: yeah that makes sense, didn't look too closely here and just copied prev logic, but assetId compare would be much safer

Comment (kaladinlight) [reply packages/asset-service/src/generateAssetData/coingecko.ts]: https://github.com/shapeshift/lib/pull/790/commits/17a287723d8250615d3e772289f7db26a1827ce9

File: packages/asset-service/src/generateAssetData/coingecko.ts
Comment (kaladinlight) [reply packages/asset-service/src/generateAssetData/coingecko.ts:53]: the string value is a const, the `as const` is intended to be used when referencing the underlying value of a string variable

```
const assetNamespace = 'erc20'
toAssetId({ chainId, assetNamespace: assetNamespace as const})
```

needed since the type of `assetNamespace` var is string, but we want the type to be inferred as the underlying constant value of `erc20`

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**
